verticapy/connection/read.py

```diff
@@ -220,7 +222,8 @@ def read_dsn(section: str, dsn: Optional[str] = None) -> dict:
             ):
                 option_val = option_val.lower()
                 conn_info[option_name] = option_val in ("true", "t", "yes", "y")
-
+            elif option_name == "oauth_config":
+                conn_info[option_name] = ast.literal_eval(option_val)
```

Comment:
Let me explain by example:
host is the option_name and dbadmin is the optional_val. 
```
conn_info = { "host" : "dbadmin" }
```

The above code only ensures that the dictionary inside oauth_config is treated as a dictionary instead of a text:

```
conn_info = {"oauth_config" : "{access_token: ...}"}